File: companion-daemon/test_ws_client.js
const WebSocket = require('ws');
const ws = new WebSocket('ws://localhost:9999/ws');

ws.on('open', function open() {
  console.log("Connected. Sending 'hello' with 100ms delays...");
  const msg = "hello";
  let i = 0;
  const interval = setInterval(() => {
    if (i >= msg.length) {
      clearInterval(interval);
      console.log("Done sending.");
      setTimeout(() => process.exit(0), 2000);
      return;
    }
    ws.send(Buffer.from([msg.charCodeAt(i)]));
    i++;
  }, 100);
});

File: companion-daemon/test_ws_client.py
import asyncio
import websockets
import sys

async def main():
    async with websockets.connect("ws://localhost:9999/ws") as websocket:
        print("Connected. Sending 'hello' with 100ms delays...")
        for char in "hello":
            await websocket.send(char.encode('utf-8'))
            await asyncio.sleep(0.1)
        print("Done sending.")
        
        # Keep connection open for a bit
        await asyncio.sleep(2)

if __name__ == "__main__":
    asyncio.run(main())
